### PR TITLE
document toml use of filterwarnings

### PR DESCRIPTION
**Pull request by graingert**
and include a demo of toml 'literal strings', which tripped me up: https://github.com/toml-lang/toml/issues/80#issuecomment-667617263

### CODE REVIEW

File: doc/en/warnings.rst [hunk 1]
```diff
@@ -78,6 +79,16 @@ all other warnings into errors.
...
         ignore::UserWarning
+        ignore:function ham\(\) is deprecated:DeprecationWarning
+
+.. code-block:: toml
+
+    [tool.pytest.ini_options]
+    filterwarnings = [
+        "error",
+        "ignore::UserWarning",
+        'ignore:function ham\(\) is deprecated:DeprecationWarning',
```
Comment (pradyunsg) [inline doc/en/warnings.rst:90]: I'd suggest adding a comment about why single quotes vs double quotes matters here. :)

Comment (graingert) [reply doc/en/warnings.rst:90]: I don't know how best to phrase it, can you make a `` ```suggestion ``, please?

**Comment by bluetech** [comment]
Looking at https://pytest--7611.org.readthedocs.build/en/7611/warnings.html, I suggest adding a comment at the top of each snippet, `# pytest.ini`, `# pyproject.toml`.

File: doc/en/warnings.rst (hunk 1 above)
Comment (pradyunsg) [reply doc/en/warnings.rst:90]: Sorry, I don't have a good suggestion as to the exact phrase either.

File: doc/en/warnings.rst [hunk 2]
```diff
@@ -78,6 +79,16 @@ all other warnings into errors.
     filterwarnings =
         error
         ignore::UserWarning
+        ignore:function ham\(\) is deprecated:DeprecationWarning
+
+.. code-block:: toml
+
+    [tool.pytest.ini_options]
+    filterwarnings = [
```
Comment (nicoddemus) [inline doc/en/warnings.rst:87]: How about:

```suggestion
    # note the use of single quote below to denote "raw" strings in TOML
    filterwarnings = [
```

**Pull request merged**